src/main/java/de/tum/in/www1/artemis/domain/settings/ide/UserIdeMappingId.java

```diff
@@ -0,0 +1,48 @@
+package de.tum.in.www1.artemis.domain.settings.ide;
+
+import java.io.Serializable;
+
+import jakarta.persistence.IdClass;
+
+import de.tum.in.www1.artemis.domain.enumeration.ProgrammingLanguage;
+
+@IdClass(UserIdeMappingId.class)
+public class UserIdeMappingId implements Serializable {
+
+    private Long user;
+
+    private ProgrammingLanguage programmingLanguage;
+
+    public UserIdeMappingId() {
+        // empty constructor for Jackson
+    }
+
+    public UserIdeMappingId(Long user, ProgrammingLanguage programmingLanguage) {
+        this.user = user;
+        this.programmingLanguage = programmingLanguage;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
The Server tests are probably failing because of the implementation here

Reply:
why does this kill the server tests?

Reply:
i removed the methods now